app/views/partials/_admin_login.html.erb

```diff
@@ -0,0 +1,7 @@
+<div class="login">
+  <%= form_for @admin, url: admin_login_path do |f| %>
+  <%= f.text_field :username, placeholder: "Username" %>
+  <%= f.password_field :password, placeholder: "Password" %>
+  <%= f.submit "Login", url: admin_login_path, method: :post %>
```

Comment:
We have url: admin_login_path twice. I think the last line only needs to be : f.submit "Login"